templates/pkg/crd_sdk.go.tpl

```diff
@@ -188,22 +189,38 @@ func (rm *resourceManager) newCreateRequestPayload(
 func (rm *resourceManager) sdkUpdate(
 	ctx context.Context,
 	r *resource,
+	diffReporter *ackreporter.Reporter,
 ) (*resource, error) {
 {{- if .CRD.Ops.Update }}
 	input, err := rm.newUpdateRequestPayload(r)
 	if err != nil {
 		return nil, err
 	}
-{{ $setCode := GoCodeSetUpdateOutput .CRD "resp" "ko.Status" 1 }}
+
+    {{ if .CRD.HasCustomUpdateOperations }}
+
+    {{ $setCode := GoCodeSetUpdateOutput .CRD "" "ko.Status" 1 }}
+    {{ GoCodeCustomUpdate .CRD "ctx" "r" "diffReporter" 1 }}
+    // Merge in the information we read from the API call above to the copy of
+    // the original Kubernetes object we passed to the function
+    ko := r.ko.DeepCopy()
+    {{ $setCode }}
+
+    {{ else }}
+
+    {{ $setCode := GoCodeSetUpdateOutput .CRD "resp" "ko.Status" 1 }}
 	{{ if and .CRD.StatusFields ( not ( Empty $setCode ) ) }}resp{{ else }}_{{ end }}, respErr := rm.sdkapi.{{ .CRD.Ops.Update.Name }}WithContext(ctx, input)
 	if respErr != nil {
 		return nil, respErr
 	}
-	// Merge in the information we read from the API call above to the copy of
-	// the original Kubernetes object we passed to the function
-	ko := r.ko.DeepCopy()
-{{ $setCode }}
+    // Merge in the information we read from the API call above to the copy of
+    // the original Kubernetes object we passed to the function
+    ko := r.ko.DeepCopy()
+    {{ $setCode }}
```

Comment:
Make sure you replace four spaces with `\t` characters when in templates :)

Reply:
Code updated.